stratum/hal/bin/barefoot/device_config_builder.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string>
+
+#include "absl/strings/str_cat.h"
+#include "gflags/gflags.h"
+#include "nlohmann/json.hpp"
+#include "p4/config/v1/p4info.pb.h"
+#include "stratum/glue/init_google.h"
+#include "stratum/glue/logging.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/glue/status/status_macros.h"
+#include "stratum/hal/lib/barefoot/bf.pb.h"
+#include "stratum/lib/macros.h"
+#include "stratum/lib/utils.h"
+
+DEFINE_string(p4c_conf_file, "",
+              "Path to the JSON output .conf file of the p4c compiler");
+DEFINE_string(bfrt_device_config_text_file, "bfrt_device_config.pb.txt",
+              "Path to text file for BfPipelineConfig output");
+DEFINE_string(bfrt_device_config_binary_file, "bfrt_device_config.pb.bin",
+              "Path to file for serialized BfPipelineConfig output");
+
+namespace stratum {
+namespace hal {
+namespace barefoot {
+
+static ::util::Status Main(int argc, char* argv[]) {
+  InitGoogle(argv[0], &argc, &argv, true);
+  InitStratumLogging();
+
+  CHECK_RETURN_IF_FALSE(!FLAGS_p4c_conf_file.empty())
+      << "p4c_conf_file must be specified.";
+
+  nlohmann::json conf;
```

Comment:
should we merge common code with new utils library?

Reply:
The helper function expects an archive string, while the builder can work with the output on disk directly.
Once we remove the tar format, this implementation will be the only one.

Reply:
Okay, makes sense.